run/hello-broken/Dockerfile

```diff
@@ -0,0 +1,48 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START run_broken_dockerfile]
+
+# Use the offical Golang image to create a build artifact.
+# This is based on Debian and sets the GOPATH to /go.
+# https://hub.docker.com/_/golang
+FROM golang:1.13 as builder
+
+# Create and change to the app directory.
+WORKDIR /app
+
+# Retrieve application dependencies.
+# This allows the container build to reuse cached dependencies.
+COPY go.* ./
+RUN go mod download
+
+# Copy local code to the container image.
+COPY . ./
+
+# Build the binary.
+RUN CGO_ENABLED=0 GOOS=linux go build -mod=readonly -v -o server
+
+# Use the official Alpine image for a lean production container.
+# https://hub.docker.com/_/alpine
+# https://docs.docker.com/develop/develop-images/multistage-build/#use-multi-stage-builds
+FROM alpine:3
+RUN apk add --no-cache ca-certificates
```

Comment:
why alpine and not distroless?

Reply:
1. As lean options go, Alpine is more generally known and "standard"
2. Alpine is the default across multi-stage Dockerfiles for Knative & Cloud Run
3. For the purpose of learning to use containers, having a shell inside by default is valuable, distroless makes an opaque box.

Happy to discuss under which circumstances (including all) we might want to switch to distroless but I'd prefer to do that outside scope of this PR.

Reply:
tbh I'm surprised Alpine is so popular for Go even though it doesn't have libc.

I'm not a fan of the `CGO_ENABLED=0` switch, because people may legitimately want to use cgo in their app (libc concern aside).

/cc @ahmetb

Reply:
Note that Debian, debian slim, and Alpine are the only distros in use in the samples across language and almost all go samples for Knative and Cloud Run currently use this. We can move individual languages as make sense with some accompanying minor docs changes.

Alternatives to the current behavior which I favor:

* Switch builder to `golang:1.13-alpine`, should be sufficient to drop CGO_ENABLED.
* Switch from `alpine:3` to `debian:buster-slim`, better alignment with developers new to containers and at 28mb still somewhat small.

Reply:
Does debian:buster-slim have CA certs? I assume not.

but at 28 MB that sounds like a great option.

Reply:
buster-slim does not.

Reply:
Here's what this looks like with comments stripped. Clocks in at 83.6 MB.

```Dockerfile
FROM golang:1.13-buster as builder
WORKDIR /app
COPY go.* ./
RUN go mod download
COPY . ./
RUN go build -mod=readonly -v -o server

FROM debian:buster-slim
RUN set -ex; \
    apt-get update; \
    DEBIAN_FRONTEND=noninteractive apt-get install -y \
      ca-certificates \
    ; \
    rm -rf /var/lib/apt/lists/*
COPY --from=builder /app/server /server
CMD ["/server"]
```

Reply:
most people will be confused by the lack of `&&`, I really recommend keeping the command the way people are used to in docker world (&&) and without `set -ex`.

Reply:
I didn't invent this, I learned it from the Docker world. That said, I will simplify as you suggest.